ancpbids/pybids_compat.py

```diff
@@ -236,13 +244,21 @@ def get_entities(self, scope: str = None, sort: bool = False) -> dict:
             result = {k: sorted(v) for k, v in sorted(result.items())}
         return result
 
-    def get_dataset_description(self) -> dict:
+    def get_dataset_description(self) -> list:
         """
         Returns
         -------
-            the dataset's dataset_description.json as a dictionary or None if not provided
+            the dataset's dataset_description.json as list of dictionaries or None if not
+            provided. if one or more derivatives datasets are available, this list will
+            contain one element per dataset, with the raw dataset always coming first.
         """
-        return self.dataset.dataset_description
+        dsets = [self]
```

Comment:
we can use a query here: self.dataset.select(self.schema.DatasetDescriptionFile).objects() which will return all dataset_description.json files as they are mapped to graph nodes of type DatasetDescriptionFile/DerivativeDatasetDescriptionFile